blocks/commerce-b2b-negotiable-quote/commerce-b2b-negotiable-quote.js

```diff
@@ -211,26 +213,50 @@ export default async function decorate(block) {
                     postcode: formValues.postcode,
                     countryCode: formValues.countryCode,
                     telephone: formValues.telephone,
-                    saveInAddressBook: formValues.saveInAddressBook,
                   };
 
                   // These values are not part of the standard address input
                   const additionalAddressInput = {
                     vat_id: formValues.vatId,
                   };
 
+                  const createCustomerAddressInput = {
+                    city: formValues.city,
+                    company: formValues.company,
+                    countryCode: formValues.countryCode,
+                    defaultBilling: !!formValues.defaultBilling || false,
+                    defaultShipping: !!formValues.defaultShipping || false,
+                    fax: formValues.fax,
+                    firstname: formValues.firstName,
+                    lastname: formValues.lastName,
+                    middlename: formValues.middlename,
+                    postcode: formValues.postcode,
+                    prefix: formValues.prefix,
+                    region: regionCode ? {
+                      regionCode,
+                      regionId: regionIdNumber,
+                    } : undefined,
+                    street: streetInputValues,
+                    suffix: formValues.suffix,
+                    telephone: formValues.telephone,
+                    vatId: formValues.vatId,
+                  };
+
                   progressSpinner.removeAttribute('hidden');
                   shippingInformation.setAttribute('hidden', true);
-                  setShippingAddress({
-                    quoteUid: quoteId,
-                    addressData: {
-                      ...addressInput,
-                      additionalInput: additionalAddressInput,
-                    },
-                  }).finally(() => {
-                    progressSpinner.setAttribute('hidden', true);
-                    shippingInformation.removeAttribute('hidden');
-                  });
+
+                  createCustomerAddress(createCustomerAddressInput)
```

Comment:
It seems that both files now always call createCustomerAddress when submitting a new address form. There's no check for whether the user wants to save to address book (by checking the checkbox). Is this expected?